docs/examples/likelihoods_guide.py

```diff
@@ -0,0 +1,263 @@
+#
+# In this notebook we will walk users through the process of creating a new likelihood
+# in GPJax.
+#
+# ## Background
+#
+# In this section we'll provide a short introduction to likelihoods and why they are
+# important. For users who are already familiar with likelihoods, feel free to skip to
+# the next section, and for users who would like more information than is provided
+# here, please see our [introduction to Gaussian processes notebook](intro_to_gps.py).
+#
+# ### What is a likelihood?
+#
+# We adopt the notation of our
+# [introduction to Gaussian processes notebook](intro_to_gps.py) where we have a
+# Gaussian process (GP) $f(\cdot)\sim\mathcal{GP}(m(\cdot), k(\cdot, \cdot))$ and a
+# dataset $\mathbf{y} = \{y_n\}_{n=1}^N$ observed at corresponding inputs
+# $\mathbf{x} = \{x_n\}_{n=1}^N$. The evaluation of $f$ at $\mathbf{x}$ is denoted by
+# $\mathbf{f} = \{f(x_n)\}_{n=1}^N$. The _likelihood function_ of the GP is then given
+# by
+# $$
+# \begin{align}
+#     \label{eq:likelihood_fn}
+#     p(\mathbf{y}\mid \mathbf{f}) = \prod_{n=1}^N p(y_n\mid f(x_n))\,.
+# \end{align}
+# $$
+# Conceptually, this conditional distribution describes the probability of the observed
+# data, conditional on the latent function values.
+#
+# ### Why is the likelihood important?
+#
+# Choosing the correct likelihood function when building a GP, or any Bayesian model
+# for that matter, is crucial. The likelihood function encodes our assumptions about
+# the data and the noise that we expect to observe. For example, if we are modelling
+# air pollution, then we would not expect to observe negative values of pollution. In
+# this case, we would choose a likelihood function that is only defined for positive
+# values. Similarly, if our data is the proportion of people who voted for a particular
+# political party, then we would expect to observe values between 0 and 1. In this
+# case, we would choose a likelihood function that is only defined for values between
+# 0 and 1.
+#
+# ## Likelihoods in GPJax
+#
+# In GPJax, all likelihoods are a subclass of the `AbstractLikelihood` class. This base
+# abstract class contains the three core methods that all likelihoods must implement:
+# `predict`, `link_function`, and `expected_log_likelihood`. We will discuss each of
+# these methods in the forthcoming sections, but first we will show how to instantiate
+# a likelihood object. To do this, we'll need a dataset.
+
+# +
+# Enable Float64 for more stable matrix inversions.
+from jax.config import config
+
+config.update("jax_enable_x64", True)
+
+import gpjax as gpx
+import jax
+import jax.numpy as jnp
+import jax.random as jr
+import matplotlib.pyplot as plt
+import tensorflow_probability.substrates.jax as tfp
+
+tfd = tfp.distributions
+plt.style.use("./gpjax.mplstyle")
+cols = plt.rcParams["axes.prop_cycle"].by_key()["color"]
+key = jr.PRNGKey(123)
+
+
+N = 50
+x = jnp.sort(jr.uniform(key=key, shape=(N, 1), minval=-3.0, maxval=3.0), axis=0)
```

Comment:
Consistency -> lower case `n` is used in the other notebooks.
```suggestion
n = 50
x = jnp.sort(jr.uniform(key=key, shape=(n, 1), minval=-3.0, maxval=3.0), axis=0)
```